### PR TITLE
feat(core): support multiple spreads on element

### PR DESCRIPTION
**Pull request by raymondmuller**
## Description
Right now Mitosis doesn't support multiple spreads ([see the example here in the fiddle](https://mitosis.builder.io/?outputTab=E4UwhgxgLkA%3D&code=JYWwDg9gTgLgBAbzgVwM4FMDKMCGN1wC%2BcAZlBCHAEQACARssADYAm6UAdMBAPQjAwIqYKioBuAFAT0AD0iw4bEjmRN4JZADsAxjG6a4AWQCeAYQqRN6TTAAUYcmFQBKRBLhxtETangBtPBgoVAAaOAwYAEEYINQAXTgAXhQMbDx0WwQAcgALdCYmCCyALjgsgHdoVizCZykPKHQYZCgDW3cPToAeACFkGO9EDmGHCCdiBGGOQODCHgA%2BDudJQiA))

This allows you to have multiple spreads by changing the logic and not using _spread, but assigning a type "spread" to the bindings.

[Working example from this branch here](https://mitosis-fiddle-git-fork-raymondmuller-feat-su-71a466-builder-io.vercel.app/?outputTab=E4UwhgxgLkA%3D&code=JYWwDg9gTgLgBAbzgVwM4FMDKMCGN1wC%2BcAZlBCHAEQACARssADYAm6UAdMBAPQjAwIqYKioBuAFAT0AD0iw4bEjmRN4JZADsAxjG6a4AWQCeAYQqRN6TTAAUYcmFQBKRBLhxtETangBtPBgoVAAaOAwYAEEYINQAXTgAXhQMbDx0WwQAcgALdCYmCCyALjgsgHdoVizCZykPKHQYZCgDW3cPToAeACFkGO9EDmGHCCdiBGGOQODCHgA%2BDudJQiA)

React:
Screenshot 2022-10-07 at 18 12 12

Vue:
Screenshot 2022-10-07 at 18 12 17

### CODE REVIEW

File: packages/core/src/types/mitosis-node.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ export type BaseNode = {
    * ```
    */
   bindings: {
-    [key: string]: { code: string; arguments?: string[] } | undefined;
+    [key: string]: { code: string; arguments?: string[]; type?: string } | undefined;
```
Comment (samijaber) [inline packages/core/src/types/mitosis-node.ts:31]: Let's be precise with `type`'s possible values:

```suggestion
    [key: string]: { code: string; arguments?: string[]; type?: 'spread' } | undefined;
```

Comment (samijaber) [inline packages/core/src/types/mitosis-node.ts:31]: We should extract the binding type here so we can import it elsewhere:

```ts
type Binding = { code: string; arguments?: string[]; type?: 'spread' }


bindings: {
  [key: string]: Binding | undefined
}
```

File: packages/core/src/generators/vue/blocks.ts [hunk 2]
```diff
@@ -283,8 +283,10 @@ const NODE_MAPPERS: {
 const stringifyBinding =
   (node: MitosisNode) =>
   ([key, value]: [string, { code: string; arguments?: string[] } | undefined]) => {
-    if (key === '_spread') {
-      return '';
+    if (node.bindings[key]?.type === 'spread') {
+      return ` v-bind="${encodeQuotes(
+        stripStateAndPropsRefs(node.bindings[key]?.code as string),
+      )}"`;
```
Comment (samijaber) [inline packages/core/src/generators/vue/blocks.ts:289]: We can export the Binding type, and use `value` here:
```ts
  ([key, value]: [string, Binding | undefined]) => {
    if (value?.type === 'spread') {
      return ` v-bind="${encodeQuotes(
        stripStateAndPropsRefs(value?.code as string),
      )}"`;
```

Comment (raymondmuller) [reply packages/core/src/generators/vue/blocks.ts]: done

File: packages/core/src/types/mitosis-node.ts (hunk 1 above)
Comment (raymondmuller) [reply packages/core/src/types/mitosis-node.ts]: done

Comment (raymondmuller) [reply packages/core/src/types/mitosis-node.ts]: done

**Review by samijaber — APPROVED** [review]

**Pull request merged**